include/NotificationsManager.php

```diff
@@ -28,109 +32,762 @@ public function __construct() {
+				$sql_seen
+			ORDER BY `item`.`created` DESC LIMIT %d, %d ",
+				intval(local_user()),
+				intval($start),
+				intval($limit)
+		);
+
+		if(dbm::is_result($r))
+			$notifs = $this->formatNotifs($r, $ident);
+
+		$arr = array (
+			'notifications' => $notifs,
+			'ident' => $ident,
+			'total' => $total,
+		);
+
+		return $arr;
+	}
+
+	/**
+	 * @brief Total number of system notifications 
+	 * @param int|string $seen
+	 *	If 0 only include notifications into the query
+	 *	which aren't marked as "seen"
+	 * @return int Number of system notifications
+	 */
+	private function systemTotal($seen = 0) {
+		if($seen === 0)
+			$sql_seen = " AND `seen` = 0 ";
+
+		$r = q("SELECT COUNT(*) AS `total` FROM `notify` WHERE `uid` = %d $sql_seen",
+			intval(local_user())
+		);
+
+		if(dbm::is_result($r))
+			return $r[0]['total'];
+
+		return 0;
+	}
+
+	/**
+	 * @brief Get system notifications
+	 * 
+	 * @param int|string $seen
+	 *	If 0 only include notifications into the query
+	 *	which aren't marked as "seen"
+	 * @param int $start Start the query at this point
+	 * @param int $limit Maximum number of query results
+	 * 
+	 * @return array with
+	 *	string 'ident' => Notification identifier
+	 *	int 'total' => Total number of available system notifications
+	 *	array 'notifications' => System notifications
+	 */
+	public function systemNotifs($seen = 0, $start = 0, $limit = 80) {
+		$ident = 'system';
+		$total = $this->systemTotal($seen);
+		$notifs = array();
+
+		if($seen === 0)
+			$sql_seen = " AND `seen` = 0 ";
+
+		$r = q("SELECT `id`, `photo`, `msg`, `date`, `seen` FROM `notify`
+				WHERE `uid` = %d $sql_seen ORDER BY `date` DESC LIMIT %d, %d ",
+			intval(local_user()),
+			intval($start),
+			intval($limit)
+		);
+
+		if(dbm::is_result($r))
+			$notifs = $this->formatNotifs($r, $ident);
+
+		$arr = array (
+			'notifications' => $notifs,
+			'ident' => $ident,
+			'total' => $total,
+		);
+
+		return $arr;
+	}
+
+	/**
+	 * @brief Addional SQL query string for the personal notifications
+	 * 
+	 * @return string The additional sql query
+	 */
+	private function _personal_sql_extra() {
+		$myurl = $this->a->get_baseurl(true) . '/profile/'. $this->a->user['nickname'];
+		$myurl = substr($myurl,strpos($myurl,'://')+3);
+		$myurl = str_replace(array('www.','.'),array('','\\.'),$myurl);
+		$diasp_url = str_replace('/profile/','/u/',$myurl);
+		$sql_extra .= sprintf(" AND ( `item`.`author-link` regexp '%s' or `item`.`tag` regexp '%s' or `item`.`tag` regexp '%s' ) ",
+			dbesc($myurl . '$'),
+			dbesc($myurl . '\\]'),
+			dbesc($diasp_url . '\\]')
+		);
+
+		return $sql_extra;
+	}
+
+	/**
+	 * @brief Total number of personal notifications 
+	 * @param int|string $seen
+	 *	If 0 only include notifications into the query
+	 *	which aren't marked as "seen"
+	 * @return int Number of personal notifications
+	 */
+	private function personalTotal($seen = 0) {
+		$sql_extra .= $this->_personal_sql_extra();
```

Comment:
Same like above, you add something to sql_extra - but this variable isn't defined before.